### PR TITLE
feat(agents): add canonical token buckets shared across harnesses

### PR DESCRIPTION
**Pull request by eugeneng04**
Adds one canonical token schema that agent harnesses map onto, and carries
the new buckets through to the result row.

    {"input", "cached", "cache_write", "reasoning", "output", "total"}   # None = unreported, never a fabricated 0

## Why

`input` means different things across providers today — some report the
uncached prompt, others the full prompt including cache hits — so raw
cross-harness token/cost comparisons are systematically skewed, and with
prompt caching on, cached tokens dominate a multi-turn run's prompt.
Reasoning tokens are also inconsistently bucketed. The anchor rule:
**`input` excludes cached tokens** (the convention usage leaderboards report).

## What's included

- **Shared schema** — `agents/result.py`: `TOKEN_BUCKETS` + `empty_tokens()`.
- **gemini CLI harness** — the terminal `result.stats` block maps to the
  canonical dict (`input = full input − cached`; `reasoning` derived from the
  total gap).
- **Row layer** — `ResultRow` gains `cachedTokens` / `reasoningTokens` /
  `cacheWriteTokens` / `totalTokens` (additive nullable fields, no
  `SCHEMA_VERSION` bump; historical rows stay valid). `normalize_tokens` reads
  the canonical keys first, with legacy aliases kept for older records.

## Verification

- Unit tests green (canonical + legacy + fall-through normalizer cases, the
  gemini `result.stats` mapping, and row round-trip with the new fields).
- `ruff check` and `ruff format --check` clean.

## Summary by CodeRabbit

* **New Features**
  * Expanded dashboard/token reporting to include cached, reasoning, cache-write, and total token counts alongside input/output.
  * Standardized canonical token breakdown for Gemini CLI stream results.
* **Bug Fixes**
  * Corrected token normalization to exclude cached prompt tokens from input and derive reasoning from token totals; clamps edge cases where cached exceeds full input.
  * Preserves “unreported vs zero” semantics by leaving missing buckets unset.
* **Tests**
  * Updated and extended unit tests to validate the new canonical 6-bucket token structure and the expanded dashboard row contract.

### CODE REVIEW

File: devops_bench/agents/result.py [hunk 1]
```diff
@@ -19,7 +19,17 @@
...
-__all__ = ["ToolCall", "AgentResult"]
+__all__: list[str] = ["AgentResult", "TOKEN_BUCKETS", "ToolCall", "empty_tokens"]
+
+# Canonical token buckets every harness maps onto: ``input`` is the non-cached
+# prompt, ``cached`` is cache-read only (cache writes go in ``cache_write``),
+# ``output`` excludes ``reasoning``, and ``total`` is the sum of all buckets.
+TOKEN_BUCKETS: tuple[str, ...] = ("input", "cached", "cache_write", "reasoning", "output", "total")
+
+
+def empty_tokens() -> dict[str, Any]:
```
Comment (itssimrank) [inline devops_bench/agents/result.py]: Can we make this strictly dict[str, int | None]?

File: devops_bench/results/normalize.py [hunk 2]
```diff
@@ -147,23 +152,33 @@ def _first_token(tokens: Mapping[str, Any], keys: tuple[str, ...]) -> int | None
     return None
 
 
-def normalize_tokens(tokens: Mapping[str, Any] | None) -> tuple[int | None, int | None]:
-    """Flatten a provider-defined token dict to ``(input_tokens, output_tokens)``.
+def normalize_tokens(
+    tokens: Mapping[str, Any] | None,
+) -> tuple[int | None, int | None, int | None, int | None, int | None, int | None]:
```
Comment (itssimrank) [inline devops_bench/results/normalize.py]: How about creating a class of NormalizedTokens for these 6 elements?

File: devops_bench/agents/cli/gemini_cli/parsing.py [hunk 3]
```diff
@@ -21,10 +21,39 @@
...
+    The CLI reports ``input_tokens`` as the *full* prompt with ``cached`` as a
+    subset, and rolls thinking tokens into ``total_tokens`` only — so canonical
+    ``input`` is the difference, and ``reasoning`` is derived from the total gap
+    (``total - full_input - output``) when every part is reported.
+    """
+    full_input = _int_or_none(stats.get("input_tokens"))
+    output = _int_or_none(stats.get("output_tokens"))
+    total = _int_or_none(stats.get("total_tokens"))
+    cached = _int_or_none(stats.get("cached"))
+    inp = full_input - cached if full_input is not None and cached is not None else full_input
```
Comment (itssimrank) [inline devops_bench/agents/cli/gemini_cli/parsing.py]: Does inp need a guardrail too to prevent it from becoming negative? or is that guaranteed to never occur?

File: devops_bench/agents/result.py (hunk 1 above)
Comment (eugeneng04) [reply devops_bench/agents/result.py]: yeah, I changed it!

File: devops_bench/results/normalize.py (hunk 2 above)
Comment (eugeneng04) [reply devops_bench/results/normalize.py]: Added a NormalizedTokens NamedTuple

File: devops_bench/agents/cli/gemini_cli/parsing.py (hunk 3 above)
Comment (eugeneng04) [reply devops_bench/agents/cli/gemini_cli/parsing.py]: I'm not sure if its guaranteed, but I added a floor of 0. Thanks for pointing this out!

**Review by janetkuo — APPROVED** [review]
/lgtm
